File: app/src/main/java/theo/base/ui/base/mvp/BasePresenter.java
package theo.base.ui.base.mvp;

public class BasePresenter<V extends BaseView>{

    private V view;

    public void attachView(V view) {
        this.view = view;
    }

    public void detachView() {
        this.view = null;
    }

    public boolean isViewAttached() {
        return this.view != null;
    }

    public V getView() {
        if (isViewAttached()) {
            return this.view;
        }

        throw new ViewNotAttachedException();
    }
}

File: app/src/main/kotlin/theo/base/ui/base/mvp/BasePresenter.kt
package theo.base.ui.base.mvp

class BasePresenter<V : BaseView> {

    private var view: V? = null

    val isViewAttached: Boolean
        get() = this.view != null

    fun attachView(view: V) {
        this.view = view
    }

    fun detachView() {
        this.view = null
    }

    fun getView(): V {
        if (isViewAttached) {
            return this.view!!
        }

        throw ViewNotAttachedException()
    }
}